fe/fe-core/src/main/java/com/starrocks/lake/snapshot/RestoreClusterSnapshotMgr.java

```diff
@@ -0,0 +1,124 @@
+        for (String arg : args) {
+            if (arg.equalsIgnoreCase("-cluster_snapshot")) {
+                LOG.info("FE start to restore from a cluster snapshot");
+                instance = new RestoreClusterSnapshotMgr(clusterSnapshotYamlFile);
+                return;
+            }
+        }
+    }
+
+    public static boolean isRestoring() {
+        return instance != null;
+    }
+
+    public static void finishRestoring() throws DdlException {
+        RestoreClusterSnapshotMgr self = instance;
+        if (self == null) {
+            return;
+        }
+
+        try {
+            List<ClusterSnapshotConfig.Frontend> frontends = self.config.getFrontends();
+            if (frontends != null) {
+                NodeMgr nodeMgr = GlobalStateMgr.getCurrentState().getNodeMgr();
+                // Drop old frontends
+                for (Frontend frontend : nodeMgr.getOtherFrontends()) {
+                    LOG.info("Drop old frontend {}", frontend);
+                    nodeMgr.dropFrontend(frontend.getRole(), frontend.getHost(), frontend.getEditLogPort());
+                }
+
+                // Add new frontends
+                for (ClusterSnapshotConfig.Frontend frontend : frontends) {
+                    LOG.info("Add new frontend {}", frontend);
+                    nodeMgr.addFrontend(frontend.isFollower() ? FrontendNodeType.FOLLOWER : FrontendNodeType.OBSERVER,
+                            frontend.getHost(), frontend.getEditLogPort());
+                }
+            }
+
+            List<ClusterSnapshotConfig.ComputeNode> computeNodes = self.config.getComputeNodes();
+            if (computeNodes != null) {
+                SystemInfoService systemInfoService = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo();
+                for (Backend be : systemInfoService.getIdToBackend().values()) {
+                    LOG.info("Drop old backend {}", be);
+                    systemInfoService.dropBackend(be.getHost(), be.getHeartbeatPort(),
+                            WarehouseManager.DEFAULT_WAREHOUSE_NAME, false);
+                }
+
+                // Drop old compute nodes
+                for (ComputeNode cn : systemInfoService.getIdComputeNode().values()) {
+                    LOG.info("Drop old compute node {}", cn);
+                    systemInfoService.dropComputeNode(cn.getHost(), cn.getHeartbeatPort(),
+                            WarehouseManager.DEFAULT_WAREHOUSE_NAME);
+                }
+
+                // Add new compute nodes
+                for (ClusterSnapshotConfig.ComputeNode cn : computeNodes) {
+                    LOG.info("Add new compute node {}", cn);
+                    systemInfoService.addComputeNode(cn.getHost(), cn.getHeartbeatServicePort(),
+                            WarehouseManager.DEFAULT_WAREHOUSE_NAME);
+                }
+            }
+
+            // TODO: Update storage volume
+
+        } finally {
+            // Rollback config
```

Comment:
Reserve the config param we used, so it can be tracked how the frontend are started.

Reply:
Because the config is modified in the constructor of the class, not the value set by user, the value of the config should be rollbacked after restoring.